### PR TITLE
multi : Godcr newui

### LINKED ISSUE #259: Implement new GUI mockups
Some mockups have been provided as an upgrade to the current godcr UI. One of the provided mockups should be selected and implemented into the godcr codebase

These are links to the sketch artboard containing the mockups

1. https://www.sketch.com/s/34b25f95-6ea5-4d3c-9a2c-f3965b8c156a/a/DG21Ea/play
2. https://www.sketch.com/s/34b25f95-6ea5-4d3c-9a2c-f3965b8c156a/a/laelKK/play

**Previews**

**1.**

**2.**

### LINKED ISSUE #264: copy blue texts to clipboard when clicked
From the new UI implementation, texts with blue colour should be clickable and they should be copied when clicked. An example can be seen in the screenshot below.

screenshot of a transaction detail page.
Screenshot 2020-11-17 at 08 50 28

### PR DESCRIPTION
**Pull request by Sirmorrison**
Fix #259 
Fix #264
Close #260 
This PR Implements the new Godcr UI Navigation, overview page and transactions page.

Transaction pages
==================
image

image

Overview Page
==================

image

image

Transaction details page
=====================
image

image

image

### CODE REVIEW

File: ui/decredmaterial/collapsible.go [hunk 1]
```diff
@@ -9,43 +9,36 @@ import (
...
 
 type Collapsible struct {
-	isExpanded            bool
-	buttonWidget          *widget.Clickable
-	line                  *Line
-	expandedIcon          *widget.Icon
-	collapsedIcon         *widget.Icon
+	isExpanded   bool
+	buttonWidget *widget.Clickable
+	// line                  *Line
```
Comment (oshorefueled) [inline ui/decredmaterial/collapsible.go]: remove commented code

File: ui/decredmaterial/collapsible.go [hunk 2]
```diff
@@ -9,43 +9,36 @@ import (
...
-		buttonWidget:          new(widget.Clickable),
+		expandIcon:            t.navMoreIcon,
+		// line:                  t.Line(),
+		buttonWidget: new(widget.Clickable),
 	}
 
-	c.line.Color = t.Color.Gray
-	c.line.Color.A = 140
+	// c.line.Color = t.Color.Gray
+	// c.line.Color.A = 140
```
Comment (oshorefueled) [inline ui/decredmaterial/collapsible.go]: remove commented code

File: ui/decredmaterial/collapsible.go [hunk 3]
```diff
@@ -59,10 +52,10 @@ func (c *Collapsible) Layout(gtx layout.Context, header func(C) D, content func(
...
 
 	dims := layout.Flex{Axis: layout.Vertical}.Layout(gtx,
-		layout.Rigid(func(gtx C) D {
-			c.line.Width = gtx.Constraints.Max.X
-			return c.line.Layout(gtx)
-		}),
+		// layout.Rigid(func(gtx C) D {
+		// 	c.line.Width = gtx.Constraints.Max.X
+		// 	return c.line.Layout(gtx)
+		// }),
```
Comment (oshorefueled) [inline ui/decredmaterial/collapsible.go]: remove commented code

File: ui/decredmaterial/combo.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+	"image"
+	"image/color"
+
+	"gioui.org/layout"
+	"gioui.org/op/paint"
+	"gioui.org/unit"
+	"gioui.org/widget"
+)
+
+type Combo struct {
```
Comment (oshorefueled) [inline ui/decredmaterial/combo.go]: I'm not sure what a "combo" means. Seems like dropdown would be a better name.

File: ui/decredmaterial/combo.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+				}
+				return layout.Dimensions{}
+			})
+		})
+	})
+}
+
+func (c *Combo) layoutOption(gtx layout.Context, itemIndex int, isFirstOption bool) layout.Dimensions {
+	btn := c.items[itemIndex].button
+
```
Comment (oshorefueled) [inline ui/decredmaterial/combo.go]: you can remove the space

File: ui/decredmaterial/combo.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+					}),
+					layout.Rigid(func(gtx C) D {
+						// if i < len(comboItemRows)-1 {
+						// 	return layout.Inset{
+						// 		Top:    unit.Dp(10),
+						// 		Bottom: unit.Dp(10),
+						// 	}.Layout(gtx, func(gtx C) D {
+						// 		return c.line.Layout(gtx)
+						// 	})
+						// }
```
Comment (oshorefueled) [inline ui/decredmaterial/combo.go]: remove commented code

File: ui/decredmaterial/combo.go [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+						return layout.Dimensions{}
+					}),
+				)
+			})
+		})
+	})
+}
+
+// drawlayout wraps the page tx and sync section in a card layout
+func (c *Combo) drawlayout(gtx layout.Context, isPopUp bool, body layout.Widget) layout.Dimensions {
```
Comment (oshorefueled) [inline ui/decredmaterial/combo.go]: use camel casing for the method name

File: ui/overview_page.go [hunk 8]
```diff
@@ -231,27 +212,34 @@ func (pg *overviewPage) recentTransactionsColumn(gtx layout.Context, c pageCommo
...
+			if txn.Status == "confirmed" {
+				txnWidgets.status.Text = formatDateOrTime(txn.Txn.Timestamp)
+				txnWidgets.statusIcon = &widget.Image{Src: paint.NewImageOp(c.icons.confirmIcon)}
+			} else {
+				txnWidgets.status.Text = txn.Status
+				txnWidgets.statusIcon = &widget.Image{Src: paint.NewImageOp(c.icons.pendingIcon)}
+			}
+
+			txnWidgets.direction.Scale = 0.07
+			txnWidgets.statusIcon.Scale = 0.03
```
Comment (oshorefueled) [inline ui/overview_page.go:234]: add a comment describing what these constants are for

**Review by oshorefueled — CHANGES_REQUESTED** [review]

File: ui/page.go [hunk 9]
```diff
@@ -108,42 +172,264 @@ func (win *Window) addPages(decredIcons map[string]image.Image) {
...
+				})
+			}),
+			layout.Rigid(func(gtx C) D {
+				l := page.theme.Line()
+				l.Color = page.theme.Color.Background
+				l.Width = gtx.Constraints.Min.X
+				l.Height = 2
+				return l.Layout(gtx)
+			}),
+		)
```
Comment (oshorefueled) [inline ui/page.go:333]: why wasn't the tab widget modified to do this?

File: ui/transaction_details_page.go [hunk 10]
```diff
@@ -41,233 +50,402 @@ func (win *Window) TransactionPage(common pageCommon) layout.Widget {
...
-				lb := common.theme.Body1(label)
-				lb.Color = common.theme.Color.Hint
-				return lb.Layout(gtx)
+				return pg.txnInfoSection(gtx, "From", transaction.WalletName, transaction.Txn.Inputs[0].AccountName, true, false)
+			}),
+			// layout.Rigid(func(gtx C) D {
+			// 	return layout.Inset{Bottom: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+			// 		return pg.txnInfoSection(gtx, "To", "", transaction.Txn.Outputs[0].Address, false, true)
+			// 	})
+			// }),
```
Comment (oshorefueled) [inline ui/transaction_details_page.go]: remove commented code

File: ui/transactions_page.go [hunk 11]
```diff
@@ -16,18 +15,19 @@ import (
 	"gioui.org/text"
 	"gioui.org/widget"
 
-	"github.com/decred/dcrd/dcrutil"
+	// "github.com/decred/dcrd/dcrutil"
```
Comment (oshorefueled) [inline ui/transactions_page.go]: remove commented code

File: ui/wallet_page.go [hunk 12]
```diff
@@ -418,9 +418,9 @@ func (pg *walletPage) Handle(common pageCommon) {
 	//	pg.subPage = subWalletMain
 	//}
 
-	if common.navTab.Selected != 1 {
-		pg.subPage = subWalletMain
-	}
+	// if common.navTab.Selected != 1 {
+	// 	pg.subPage = subWalletMain
+	// }
```
Comment (oshorefueled) [inline ui/wallet_page.go]: remove commented code

**Review by oshorefueled — CHANGES_REQUESTED** [review]

File: ui/page.go (hunk 9 above)
Comment (Sirmorrison) [reply ui/page.go:333]: No idea. This can be revisited when the PR is merged to prevent further stalling in the project.

**Review by oshorefueled — APPROVED** [review]

**Pull request merged**